api/analyticsd/service/service.go

```diff
@@ -0,0 +1,224 @@
+		return &pb.IdentifyResponse{}, nil
+	}
+	traits := segment.NewTraits()
+	traits.Set("account_type", req.AccountType)
+	traits.Set(s.prefix+"signup", "true")
+	if req.Email != "" {
+		traits.SetEmail(req.Email)
+	}
+	for key, value := range req.Properties.AsMap() {
+		traits.Set(key, value)
+	}
+	err := s.segment.Enqueue(segment.Identify{
+		UserId: req.Key,
+		Traits: traits,
+		Context: &segment.Context{
+			Extra: map[string]interface{}{
+				"active": req.Active,
+			},
+		},
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "enqueuing segment identify: %v", err)
+	}
+	return &pb.IdentifyResponse{}, nil
+}
+
+func (s *Service) Track(ctx context.Context, req *pb.TrackRequest) (*pb.TrackResponse, error) {
+	if req.AccountType == pb.AccountType_ACCOUNT_TYPE_USER {
+		return &pb.TrackResponse{}, nil
+	}
+	props := segment.NewProperties()
+	for key, value := range req.Properties.AsMap() {
+		props.Set(key, value)
+	}
+
+	log.Infof("enqueing event %v with props %v", eventToString(req.Event), props)
+
+	if err := s.segment.Enqueue(segment.Track{
+		UserId:     req.Key,
+		Event:      eventToString(req.Event),
+		Properties: props,
+		Context: &segment.Context{
+			Extra: map[string]interface{}{
+				"active": req.Active,
+			},
+		},
+	}); err != nil {
+		log.Errorf("calling segment enqueue: %v", err)
+		return nil, status.Errorf(codes.Internal, "enqueuing segment track: %v", err)
+	}
+
+	orgKey := ""
+	devKey := ""
+	if req.AccountType == pb.AccountType_ACCOUNT_TYPE_ORG {
+		orgKey = req.Key
+		devKey, _ = req.Properties.AsMap()["member"].(string)
+	} else if req.AccountType == pb.AccountType_ACCOUNT_TYPE_DEV {
+		devKey = req.Key
+		orgKey, _ = req.Properties.AsMap()["org_key"].(string)
+	}
+
+	// Don't process the events that are generated by s.frc.ProcessAnalyticsEvent
+	// when a reward or claim is created and sent here for analytics tracking.
+	// Also be sure that we at least have an org key since that is required.
+	if s.frc != nil && req.Event != pb.Event_EVENT_FIL_REWARD && req.Event != pb.Event_EVENT_FIL_CLAIM && req.Event != pb.Event_EVENT_FIL_FINALIZE_CLAIM && orgKey != "" {
+		var opts []filrewards.ProcessAnalyticsEventOption
+		if devKey != "" {
+			opts = append(opts, filrewards.ProcessAnalyticsEventDevKey(devKey))
+		}
+		if _, err := s.frc.ProcessAnalyticsEvent(ctx, orgKey, req.Event, opts...); err != nil {
+			return nil, status.Errorf(codes.Internal, "processing analytics event for filrewards: %v", err)
+		}
+	}
+
+	return &pb.TrackResponse{}, nil
+}
+
+func (s *Service) Close() {
+	if err := s.segment.Close(); err != nil {
+		log.Errorf("closing segment client: %s", err)
+	}
+	log.Info("segment client disconnected")
+
+	stopped := make(chan struct{})
+	go func() {
+		s.server.GracefulStop()
+		close(stopped)
+	}()
+	t := time.NewTimer(10 * time.Second)
+	select {
+	case <-t.C:
+		s.server.Stop()
+	case <-stopped:
+		t.Stop()
+	}
+	log.Info("gRPC server stopped")
+}
+
+func eventToString(e pb.Event) string {
```

Comment:
`e.String()` wouldn't cut it here?

Reply:
It could for sure. The string would look like `EVENT_BILLING_SETUP`. I just kept the strings that @andrewxhill was already using. Any options @andrewxhill?

Reply:
for external apis, camelcase or snake is fine. there are some already being collected that are snake though, so would be nice to maintain that unless it's a real drag.

Reply:
I guess using `String()` outputs upper snake case. We could also convert those strings to all lower case. I guess the main issue/question is that the resulting string is slightly different than what we're already collecting. Is it important to keep the exact event names we are already collecting? If not, it would be nice to not have to maintain manually this map of events to names.

Reply:
keeping the same would be nice, but if we are going to change them... now is the time for sure. so just do whatever makes sense here and we'll see

Reply:
but yeah, avoiding changes going forward (or now) will be helpful

Reply:
Ok decided to go with the lower case version of `s.String()` which outputs stuff like `event_billing_setup`. Lower snake case and we don't have to maintain the manual lookup and we make the change now, not later.